concordium-contracts-common/src/schema.rs

```diff
@@ -101,6 +110,32 @@ pub struct ContractV2 {
     pub receive: BTreeMap<String, FunctionV2>,
 }
 
+/// Describes all the schemas of a V1 smart contract with a V3 schema.
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+/// The [Default] instance produces an empty schema.
+pub struct ContractV3 {
+    pub init:    Option<FunctionV3>,
+    pub receive: BTreeMap<String, FunctionV3>,
```

Comment:
This means that each function can also have an event schema. What is meant to be the meaning of this if both the init function has the event schema, and the receive function has it?

Was not the idea that the contract itself has an event schema, not the individual entrypoints. In that case, would not 
```
pub struct ContractV3 {
   pub init: ...
   pub receive: ...
   pub event: Option<Type>
}
```
be the more appropriate modelling of that?